src/main/java/com/google/api/gax/core/GaxPropertiesProvider.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2017, Google Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.api.gax.core;
+
+import com.google.api.gax.grpc.InstantiatingChannelProvider;
+import java.io.IOException;
+import java.util.Properties;
+
+/**
+ * Provides meta-data properties stored in gax.properties
+ */
+public class GaxPropertiesProvider {
+
+  private static final Properties gaxProperties = new Properties();
+  private static final String DEFAULT_VERSION = "";
+
+  /**
+   * Returns the current version of GAX
+   */
+  public static String getGaxVersion() {
+    String gaxVersion = loadGaxProperty("version");
+    return gaxVersion != null ? gaxVersion : DEFAULT_VERSION;
+  }
+
+  /**
+   * Returns the current version of gRPC
+   */
+  public static String getGrpcVersion() {
+    String grpcVersion = loadGaxProperty("grpc_version");
+    return grpcVersion != null ? grpcVersion : DEFAULT_VERSION;
+  }
+
+  private static String loadGaxProperty(String key) {
+    try {
+      if (gaxProperties.isEmpty()) {
+        gaxProperties.load(
+            InstantiatingChannelProvider.class
```

Comment:
You shouldn't depend on a class from gax.grpc in gax.core. Anyway, use GaxPropertiesProvider.class here.